lang/java/avro/src/main/java/org/apache/avro/reflect/ReflectData.java

```diff
@@ -181,13 +181,15 @@ protected boolean isRecord(Object datum) {
 
   /**
    * Returns true also for non-string-keyed maps, which are written as an array
-   * of key/value pair records.
+   * of key/value pair records. Returns false for array of bytes, since it should
```

Comment:
I find it a bit strange that the javadoc comment only documents the exception cases and not the main behavior. I would suggest the following instead:

> Returns true for arrays and false otherwise, with the following exceptions:
> * Returns true for non-string-keyed maps, which are written as an array of key/value pair records.
> * Returns false for arrays of bytes, since those should be treated as byte data type instead.